### PR TITLE
Prepare UI tests for rust 1.55

### PR DESCRIPTION
**Pull request by bkchr**
Closes: https://github.com/paritytech/substrate/issues/9627

polkadot companion: https://github.com/paritytech/polkadot/pull/3807

### CODE REVIEW

**Review by gilescope — APPROVED** [review]
LGTM. Esteban's been at it again improving those error messages - nice.

**Review by rcny — APPROVED** [review]

**Comment by bkchr** [comment]
@rcny you need to run the CI jobs of this pr with the new CI image.

@TriplEight should be able to help you with this.

**Comment by TriplEight** [comment]
I'm testing the same on the polkadot companion, then revert both commits, merge the new CI image and merge these both PRs.

**Comment by TriplEight** [comment]
Pipelines are failing with new Rust versions.

**Comment by thiolliere** [comment]
the failing message is the one of the failing job of this PR ?
If so I think this PR should fix it https://github.com/paritytech/substrate/pull/9747

**Comment by bkchr** [comment]
> the failing message is the one of the failing job of this PR ?
> If so I think this PR should fix it #9747

Nice. I will merge master after your pr is in master

File: client/service/src/client/wasm_override.rs
Comment (thiolliere) [inline client/service/src/client/wasm_override.rs:110]: I don't know much about this part of the code, but having the generic here could ensure that the same type is used every time any of its method is called. Is it required here ?

File: primitives/state-machine/src/lib.rs
Comment (thiolliere) [inline primitives/state-machine/src/lib.rs:68]: shouldn't we have the commas here optional, the one after message and the last one, maybe we can write:
```suggestion
	($message:expr $( , $arg:expr )* $( , )?) => {
```
If it works we can use the same trick in the 2 match with `target` above and merge them into: `target: $target:expr, $message:expr $( , $arg:ident )* $( , )?`

EDIT: anyway everybody should use log instead.

File: primitives/state-machine/src/lib.rs
Comment (thiolliere) [inline primitives/state-machine/src/lib.rs:82]: same maybe if my suggestion works we can do same as `warn`
```suggestion
	(target: $target:expr, $message:expr $( , $arg:ident ) * $(,)? ) => {
```

File: primitives/state-machine/src/lib.rs
Comment (thiolliere) [inline primitives/state-machine/src/lib.rs:96]: doesn't seem right needed to me as expansion isn't modified

**Review by thiolliere — COMMENTED** [review]
looks good to me,

some nits

**Review by thiolliere — APPROVED** [review]

File: client/service/src/client/wasm_override.rs
Comment (bkchr) [reply client/service/src/client/wasm_override.rs:110]: Not really, we just need the executor at startup to get the runtime versions.

**Comment by bkchr** [comment]
@thiolliere could you merge master please. 

@TriplEight after that we should be finally ready to merge this

**Comment by thiolliere** [comment]
it should be good now

**Comment by ordian** [comment]
bot merge

**Pull request merged**
